docs/docs/life-and-times-of-a-gatsby-build.md

```diff
@@ -2,7 +2,305 @@
+
+15. `onPreExtractQueries`
+
+Calls the [`onPreExtractQueries` API](/docs/node-apis/#onPreExtractQueries) for your site and all plugins implementing it.
+
+16. `update schema`
+
+Rebuilds the GraphQL schema, this time with `SitePage` context.
+
+17. `extract queries from components`
+
+All JavaScript files in the site are loaded and Gatsby determines if there are any GraphQL queries in them. If there are problematic queries they can be reported back with warnings or errors. All these queries get queued up for execution in a later step.
+
+18. `write out requires`
+
+An internal Gatsby utility adds the code that files need to load/require.
+
+19. `write out redirect data`
+
+An internal Gatsby utility that adds code for redirects, like implemented with [`createRedirect`](/docs/actions/#createRedirect).
+
+20. `Build manifest and related icons` - (from `gatsby-plugin-manifest`)
+
+This step is actually activated by `gatsby-plugin-manifest` in the `gatsby-default-starter` and is not a part of the built-in Gatsby functionality, demonstrating plugins are able to tap into the lifecycle.
+
+21. `onPostBootstrap`
+
+Calls the [`onPostBootstrap` API](/docs/node-apis/#onPostBootstrap) for your site and all plugins implementing it.
+
+### Steps of the build phase
+
+1. `run static queries`
+
+Static queries that were queued up earlier from query extraction are actually run so the data pages need can be provided to them.
+
+2. `Generating image thumbnails — 6/6` - (from `gatsby-plugin-sharp`)
+
+Another step that is not a part of the built-in Gatsby functionality, but is the result of installing `gatsby-plugin-sharp`, which taps into the lifecycle. Sharp runs processing on images to create thumbnails of different sizes.
+
+3. `Building production JavaScript and CSS bundles`
+
+Compiles JavaScript and CSS using webpack.
+
+4. `Rewriting compilation hashes`
+
+Compilation hashes are used by webpack, and all files with page data need to be updated with the new hashes since they've been recompiled.
+
+5. `run page queries`
+
+Page queries that were queued up earlier from query extraction are actually run so the data pages need can be provided to them.
+
+6. `Building static HTML for pages`
+
+With everything ready for the HTML pages in place, HTML is rendered and written out to files so it can be served up statically. Since HTML is being rendered in a server environment, [references to browser APIs like window can break the build](/docs/debugging-html-builds/).
+
+## What do you get from a successful build?
+
+When a Gatsby build is successfully completed, everything you need to deploy your site ends up in the `public` folder at the root of the site. Included in it are minified files, transformed images, JSON files with information and data for each page, static HTML pages for each page, and more.
+
+The final build is just static files so it can now be [deployed](/docs/deploying-and-hosting/).
```

Comment:
```suggestion
The final build is just static files so it can now be [deployed](/docs/deploying-and-hosting/).

You can preview the static files with `gatsby serve` which serves the files from the `public folder`.
```